006-strimzi-ui.md

```diff
@@ -0,0 +1,187 @@
+# Add a UI to Strimzi for Kafka Administration
+
+This proposal is to add a Kafka administration UI to the Strimzi project.
+
+## Motivation
+
+Kafka out of the box provides a number of useful command tools to perform common administrative tasks such as creating topics, viewing consumer group status, and consuming record content. While these are helpful, they are not always the simplest or most convenient way of monitoring and performing administrative tasks on a Kafka cluster. In these cases a dedicated administration UI, focused on common administration tasks, would not only enable a user to complete their task more quickly and simply than previously possible, but also enable a more positive user experience of Kafka and its capabilities.
+
+## Proposal
+
+This proposal will describe a Kafka administration UI capability. It will describe example capabilities which could be offered, cover at a high level how the UI code itself could be implemented (with more low level detail to be provided as required), how this could be delivered, as well as possible future work and how this could integrate with Strimzi today.
+
+### UI implementation
+
+At a high level, I would propose a UI implemented as follows:
+
+- It would be a Javascript based UI (using [babel](https://babeljs.io/) to provide latest ECMAScript capabilities in a cross browser compatible manner), using the [React](https://reactjs.org/) framework
```

Comment:
Absolutley - I havent used TypeScript in anger yet, but would be very interested to see how a Typescript UI (vs Javascript) would work and be written. Would it be possible to explore a TS version of the prototype we have been looking at?

Reply:
Ok cool! The good thing is since TS is a superset of JS, valid JS is valid TS, so there's immediately a lot of overlap and all your JS knowledge is 100% relevant. Depending on how strictly we configure `tsconfig.json`, more or less of the typing could be required. Point being, we don't have to jump right into the deep end with full `strict` mode (though its good practice). We can keep it flexy while everyone is learning as not to impede the workflow.

Which prototype were you referring to?

Reply:
Sorry - the view layer one shown at Friday's playback - you should have a copy of it. I would be very interested to see what you would reccomend configuration wise, the syntax required as a result, and discuss pros and cons of TS and JS.